src/App.js

```diff
@@ -8,7 +8,13 @@ import SubHeader from "./components/SubHeader";
 import TopNav from "./components/TopNav";
 
 function App(props) {
-    const product = props.state.products[0];
+    // const product = props.state.products[0];
+
+  const filteredProducts = props.state.products.filter((category) => {
+    return props.currentCategory
```

Comment:
the function you pass into the filter function must return true or false always. That is the rule.
you are returning currentCategory which is a string